services/nexus/nexus.service.js

```diff
@@ -1,124 +1,229 @@
-          (err, res, buffer) => {
-            if (err != null) {
-              badgeData.text[1] = 'inaccessible'
-              sendBadge(format, badgeData)
-              return
-            } else if (res && res.statusCode === 404) {
-              badgeData.text[1] = 'no-artifact'
-              sendBadge(format, badgeData)
-              return
-            }
-            try {
-              const parsed = JSON.parse(buffer)
-              let version = '0'
-              switch (repo) {
-                case 'r':
-                  if (parsed.data.length === 0) {
-                    badgeData.text[1] = 'no-artifact'
-                    sendBadge(format, badgeData)
-                    return
-                  }
-                  version = parsed.data[0].latestRelease
-                  break
-                case 's':
-                  if (parsed.data.length === 0) {
-                    badgeData.text[1] = 'no-artifact'
-                    sendBadge(format, badgeData)
-                    return
-                  }
-                  // only want to match 1.2.3-SNAPSHOT style versions, which may not always be in
-                  // 'latestSnapshot' so check 'version' as well before continuing to next entry
-                  parsed.data.every(artifact => {
-                    if (isNexusSnapshotVersion(artifact.latestSnapshot)) {
-                      version = artifact.latestSnapshot
-                      return
-                    }
-                    if (isNexusSnapshotVersion(artifact.version)) {
-                      version = artifact.version
-                      return
-                    }
-                    return true
-                  })
-                  break
-                default:
-                  version = parsed.data.baseVersion || parsed.data.version
-                  break
-              }
-              if (version !== '0') {
-                badgeData.text[1] = versionText(version)
-                badgeData.colorscheme = versionColor(version)
-              } else {
-                badgeData.text[1] = 'undefined'
-                badgeData.colorscheme = 'orange'
-              }
-              sendBadge(format, badgeData)
-            } catch (e) {
-              badgeData.text[1] = 'invalid'
-              sendBadge(format, badgeData)
-            }
-          }
-        )
+    if (repo === 'r' || repo === 's') {
+      schema = searchApiSchema
+      url += 'service/local/lucene/search'
+    } else {
+      schema = resolveApiSchema
+      url += 'service/local/artifact/maven/resolve'
+      options.qs.r = repo
+      options.qs.v = 'LATEST'
+    }
+
+    if (queryOpt) {
+      const opts = queryOpt.split(':')
+      opts.forEach(opt => {
+        const kvp = opt.split('=')
+        const key = kvp[0]
+        const val = kvp[1]
+        options.qs[key] = val
       })
-    )
+    }
+
+    if (serverSecrets) {
+      if (serverSecrets.nexus_base64auth) {
```

Comment:
Figured we could allow folks using a self-hosted shields instance the option to put their own base64 encoded nexus credentials, or username/password